### PR TITLE
Feature/blb 315 cutscene system

### PR DESCRIPTION
**Pull request by xcrowy**
- Tutorial System
- Librarian NPC Asset Integration + Animation
- Cutscene System (2/12 complete)
- Bugfixes for Lore NPC

### CODE REVIEW

File: Assets/Scripts/Controllers/Cutscene/CutsceneDialogue.cs [hunk 1]
```diff
@@ -0,0 +1,232 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+using UnityEngine.InputSystem;
+using UnityEngine.UI;
+
+public class CutsceneDialogue : MonoBehaviour
```
Comment (FishError) [inline Assets/Scripts/Controllers/Cutscene/CutsceneDialogue.cs:8]: so this seems kinda similar to normal dialogue. How come you didnt inherit from Dialogue?

File: Assets/Scripts/Tutorial/TutorialEquipmentDrop.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    public TypeOfEquipments equipments;
+    private GameObject blobEquipments;
+    private Transform[] typesOfEquipment;
+    private Equipment getEquipments;
+
+    private void OnTriggerEnter2D(Collider2D collision)
+    {
+        if (collision.gameObject.tag == "Player")
+        {
+            foreach (Transform equipment in typesOfEquipment)
```
Comment (FishError) [inline Assets/Scripts/Tutorial/TutorialEquipmentDrop.cs:22]: can you switch this to a for (int i; typesOfEquipment.Count ...
cause i actually realized recently that foreach is more resource heavy than a regular for loop lol

Comment (xcrowy) [reply Assets/Scripts/Tutorial/TutorialEquipmentDrop.cs]: done

**Review by FishError — APPROVED** [review]

File: Assets/Scripts/Tutorial/TutorialCutscene.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class TutorialCutscene : MonoBehaviour
+{
+    public void StartTutorial()
+    {
+        CutsceneSystem.Instance.CutsceneEnded();
+    }
```
[the remark this answers is not in the document]
Comment (xcrowy) [reply Assets/Scripts/Tutorial/TutorialCutscene.cs:11]: no, this is for sending a signal to the timeline animation, basically saying when the animation has reached this frame, do this function, so in this case, when the cutscene ends, we will start the tutorial, hence the CutsceneEnded

File: Assets/Scripts/Controllers/Cutscene/CutsceneDialogue.cs [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+        index = 0;
+
+        if (!personTwoEnded)
+        {
+            isPersonTwoTalking = true;
+            StartPersonTwoDialogue();
+        }
+        else
+            EndDialogue();
+    }
```
[the remark this answers is not in the document]
Comment (xcrowy) [reply Assets/Scripts/Controllers/Cutscene/CutsceneDialogue.cs:218]: i could, but thats refactoring which i dont want to do right now, cuz the whole dialogue system needs to be refactor in a later date

**Review by ashabid03 — APPROVED** [review]

**Pull request merged**